tests/tests.rs

```diff
@@ -58,11 +71,199 @@ fn test_commit_all() {
 
     File::create(dir.as_ref().join("somefile")).unwrap();
 
-    repo.add(vec!("somefile")).unwrap();
+    repo.add(vec!["somefile"]).unwrap();
 
     repo.commit_all("some commit message").unwrap();
-    
-    let output = Command::new("git").current_dir(&dir).args(&["log"]).output().unwrap();
 
-    assert!(str::from_utf8(&output.stdout).unwrap().contains("some commit message"));
-}
\ No newline at end of file
+    let output = Command::new("git")
+        .current_dir(&dir)
+        .args(&["log"])
+        .output()
+        .unwrap();
+
+    assert!(str::from_utf8(&output.stdout)
+        .unwrap()
+        .contains("some commit message"));
+}
+
+#[test]
+#[should_panic]
```

Comment:
Perhaps it might be clearer if rather than relying on the test panicing for an implicit assertion we have an explicit one around the unix status codes returned by the subprocess?

Reply:
```rust
let result = repo.remove(...); //Result<...>

assert!(result.is_err());
```

Or maybe something like this. This way everything is explicit and there is no need for should_panic.